metadata-ingestion/src/datahub/utilities/file_backed_collections.py

```diff
@@ -1,87 +1,162 @@
+            # Because we're only using these dbs to offload data from memory, we don't need
+            # to worry about data integrity too much.
+            conn.execute('PRAGMA locking_mode = "EXCLUSIVE"')
+            conn.execute('PRAGMA synchronous = "OFF"')
+            conn.execute('PRAGMA journal_mode = "MEMORY"')
+            conn.execute(f"PRAGMA journal_size_limit = {100 * 1024 * 1024}")  # 100MB
+
+            self._ref_count[filename] = 0
+            self._sqlite_connection_cache[filename] = conn
+
+        self._ref_count[filename] += 1
+        return self._sqlite_connection_cache[filename]
+
+    def drop_connection(self, filename: pathlib.Path) -> None:
+        self._ref_count[filename] -= 1
+
+        if self._ref_count[filename] == 0:
+            # Cleanup the connection object.
+            self._sqlite_connection_cache[filename].close()
+            del self._sqlite_connection_cache[filename]
+            del self._ref_count[filename]
+
+
+_sqlite_connection_cache = _SqliteConnectionCache()
+
+
 @dataclass(eq=False)
 class FileBackedDict(MutableMapping[str, _VT], Generic[_VT]):
-    """A dictionary that stores its data in a temporary SQLite database.
-
+    """
+    A dict-like object that stores its data in a temporary SQLite database.
     This is useful for storing large amounts of data that don't fit in memory.
 
-    For performance, implements an in-memory LRU cache using an OrderedDict,
-    and sets a generous journal size limit.
+    This class is not thread-safe.
     """
 
+    filename: pathlib.Path
+
     serializer: Callable[[_VT], SqliteValue]
     deserializer: Callable[[Any], _VT]
 
-    filename: str = field(default_factory=tempfile.mktemp)
-    cache_max_size: int = field(default=2000)
-    cache_eviction_batch_size: int = field(default=200)
+    tablename: str = field(default=_DEFAULT_TABLE_NAME)
+    extra_columns: Dict[str, Callable[[_VT], SqliteValue]] = field(default_factory=dict)
+
+    cache_max_size: int = field(default=_DEFAULT_MEMORY_CACHE_MAX_SIZE)
+    cache_eviction_batch_size: int = field(
+        default=_DEFAULT_MEMORY_CACHE_EVICTION_BATCH_SIZE
+    )
 
     _conn: sqlite3.Connection = field(init=False, repr=False)
+
+    # To improve performance, we maintain an in-memory LRU cache using an OrderedDict.
     _active_object_cache: OrderedDict[str, _VT] = field(init=False, repr=False)
 
     def __post_init__(self) -> None:
-        self._conn = sqlite3.connect(self.filename, isolation_level=None)
+        assert (
+            self.cache_eviction_batch_size > 0
+        ), "cache_eviction_batch_size must be positive"
+
+        self._conn = _sqlite_connection_cache.get_connection(self.filename)
 
         # We keep a small cache in memory to avoid having to serialize/deserialize
         # data from the database too often. We use an OrderedDict to build
         # a poor-man's LRU cache.
         self._active_object_cache = collections.OrderedDict()
```

Comment:
Do we need to self-manage an LRU cache?

Reply:
in this case yes it is necessary to self-manage